src/Stack.spec.ts

```diff
@@ -6,65 +6,73 @@ describe("Stack", () => {
-        const def = () => {
+    describe("ability to take defaultOptions", () => {
+        it("Should accept function which returns defaultOptions", () => {
+            const def = () => {
+                const headers: IRequestHeaders = {
+                    "Accept": "",
+                    "Authorization": "",
+                    "Content-Type": "application/json",
+                    "Accept-Language": "en"
+                };
+                return {
+                    baseUrl: "http://api.example.com/example",
+                    body: JSON.stringify({user: 1}),
+                    method: "POST",
+                    queryParameters: {user: "1", id: "2"},
+                    url: "/products",
+                    headers
+                } as IRequest;
+            };
+            const stack = new Stack(def);
+            expect(stack.defaultOptions.baseUrl).toBe("http://api.example.com/example");
+            expect(stack.defaultOptions.method).toBe("POST");
+            expect(stack.defaultOptions.url).toBe("/products");
+        });
+        it("Should accept defaultOptions as parameter", () => {
             const headers: IRequestHeaders = {
                 "Accept": "",
                 "Authorization": "",
                 "Content-Type": "application/json",
                 "Accept-Language": "en"
             };
-            return {
-                baseUrl: "http://api.example.com/example",
+            const defaultOption: IRequest = {
+                baseUrl: "api.example.com/example",
                 body: JSON.stringify({user: 1}),
                 method: "POST",
                 queryParameters: {user: "1", id: "2"},
                 url: "/products",
                 headers
-            } as IRequest;
-        };
-        const stack = new Stack(def);
-        expect(stack.defaultOptions).toBeDefined();
-    });
-    it("Should accept function which returns defaultOptions", () => {
-        const headers: IRequestHeaders = {
-            "Accept": "",
-            "Authorization": "",
-            "Content-Type": "application/json",
-            "Accept-Language": "en"
-        };
-        const defaultOption: IRequest = {
-            baseUrl: "http://api.example.com/example",
-            body: JSON.stringify({user: 1}),
-            method: "POST",
-            queryParameters: {user: "1", id: "2"},
-            url: "/products",
-            headers
-        };
+            };
 
-        const stack = new Stack(defaultOption);
-        expect(stack.defaultOptions).toBeDefined();
+            const stack = new Stack(defaultOption);
+            expect(stack.defaultOptions).toBeDefined();
+            expect(stack.defaultOptions.baseUrl).toBe("api.example.com/example");
+            expect(stack.defaultOptions.method).toBe("POST");
+            expect(stack.defaultOptions.url).toBe("/products");
+        });
     });
+    describe("", () => {
```

Comment:
Don't do a wrap without a title. The least you can do (and in most cases that will be perfectly fine and enough) is to use the method's name aka

`describe("process()", () => { .... });`

then, describing it in more detail.

Remember that we're basically writing readable sentences here. Somethings like
"Fetch process() accepts an options-parameter", "Fetch process() has a default implementation", "Fetch process() is defined", etc. read nicely, are easy to understand and will help you identify errors in case they are failing.

Also i'm personally not a big fan of "should", because it's just an extra word without an actual benefit, but i agree its not hurting anyone either. We can discuss this tuesday as well (and i know i'm not flawless here too^^)